### DISCUSSION TITLE
I can't upload to my bucket

### DISCUSSION THREAD

**Discussion by rcapdepaula** [supabase/supabase#5191] [Questions] [answered]
I'm trying to upload an image to my bucket but I get error 400.

My function:

```
const handleFilesUpload = async (e) => {
		const image = e.target.files[0]
		console.log(image)
		const { data, error } = await supabase.storage
			.from('images')
			.upload(`public/${image.name}`, image, {
				cacheControl: '3600',
				upsert: false
			})
	}
```
My input:
```

	<label for="images">Upload Your Profile Picture</label>
	<input type="file" id="images" accept=".jpg,.png,.jpeg" on:change={handleFilesUpload} />
```
I need help!

**Comment by GaryAustin1** [comment, marked as answer]
Are your object RLS policies set up correctly, especially for insert?

**Reply by rcapdepaula** [reply]
Hi Gary, thank you for your reply.  RLS is enabled. I don't set any police.

**Reply by GaryAustin1** [reply]
You have to have insert policy on your objects in images, even if just true. Public only means it can be read by everyone.

**Reply by rcapdepaula** [reply]
I did it: [https://tkiwiuuduhqbchcuyjqc.supabase.in/storage/v1/object/public/images/public/Captura%20de%20tela%202022-01-27%20163631.png](https://tkiwiuuduhqbchcuyjqc.supabase.in/storage/v1/object/public/images/public/Captura%20de%20tela%202022-01-27%20163631.png)

**Reply by GaryAustin1** [reply]
You only have a SELECT policy.  You need an INSERT policy for upload.
I normally put my policies in the top section under the section you have with "images".  I can't comment on not having anything there and only in the bottom part of the UI, I just have not looked into that level of detail.
Remember a public bucket is readable no matter the policy if someone has the url.

**Reply by rcapdepaula** [reply]
I did it :

`((bucket_id = 'images'::text) AND ((storage.foldername(name))[1] = 'public'::text) AND (role() = 'authenticated'::text))`

and Works! 

Thank you very much for your time! 

I have another doubt related to upload. In my table, can I use "add foreign key relation" to get the image URL, or do I have to use another approach?

**Reply by GaryAustin1** [reply]
You should close this question out and ask a new one here or over at Discord.
I can't help you with foreign keys as I have not needed them yet.

**Reply by rcapdepaula** [reply]
Ok Thank you!
